src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_hallsScripts.cpp

```diff
@@ -0,0 +1,298 @@
+    }
+
+    void ReceiveAIEvent(AIEventType eventType, Unit* /*sender*/, Unit* invoker, uint32 /*miscValue*/) override
+    {
+        if (eventType == AI_EVENT_CUSTOM_A) // Attacked by Player
+        {
+            if (Creature* partner = m_creature->GetMap()->GetCreature(m_sparringPartner))
+            {
+                m_creature->getThreatManager().modifyThreatPercent(partner, -101);
+                m_creature->ClearTemporaryFaction();
+                m_eventStarted = false;
+                m_creature->SetInCombatWithZone();
+            }
+        }
+        else if (eventType == AI_EVENT_CUSTOM_B) // End Duel
+        {
+            m_creature->AttackStop();
+            m_creature->ClearTemporaryFaction();
+            m_creature->HandleEmote(EMOTE_ONESHOT_BOW);
+            ResetTimer(GLADIATIOR_EVENT_RP, 1000);
+            m_eventStarted = false;
+        }
+        else if (eventType == AI_EVENT_CUSTOM_C) // End Duel
+        {
+            m_creature->SetFactionTemporary(1692, TEMPFACTION_RESTORE_RESPAWN | TEMPFACTION_RESTORE_COMBAT_STOP);
+            AttackStart(invoker);
+            m_sparringPartner = invoker->GetObjectGuid();
+        }
+    }
+    void ExecuteAction(uint32 action) override
+    {
+        switch (action)
+        {
+        case GLADIATOR_STOP_EVENT:
+        {
+            if (m_creature->GetHealthPercent() <= 40.f)
+            {
+                if (Creature* centurion = GetClosestCreatureWithEntry(m_creature, NPC_SHATTERED_HAND_CENTURION, 15.f, true, false, true))
+                    SendAIEvent(AI_EVENT_CUSTOM_A, m_creature, centurion);
+            }
+            break;
+        }
+        }
+    }
+    void UpdateAI(const uint32 diff) override
+    {
+        CombatAI::UpdateAI(diff);
+        if (m_creature->IsInCombat() && m_eventStarted)
+        {
+            if (m_creature->SelectAttackingTarget(ATTACKING_TARGET_TOPAGGRO, 0, nullptr, SELECT_FLAG_PLAYER))
+            {
+                if (Creature* partner = m_creature->GetMap()->GetCreature(m_sparringPartner))
+                {
+                    SendAIEvent(AI_EVENT_CUSTOM_A, m_creature, partner);
+                    m_creature->getThreatManager().modifyThreatPercent(partner, -101);
+                    m_creature->ClearTemporaryFaction();
+                    m_eventStarted = false;
+                }
+            }
+        }
+    }
+};
+
+enum CenturionActions
+{
+    CENTURION_EVENT_RP,
+    CENTURION_ACTION_MAX,
+    CENTURION_START_EVENT,
+};
+
+struct npc_shattered_hand_centurion : public CombatAI
+{
+    npc_shattered_hand_centurion(Creature* creature) : CombatAI(creature, CENTURION_ACTION_MAX),
+        m_instance(static_cast<ScriptedInstance*>(creature->GetInstanceData())),
+        m_eventStarted(false)
+    {
+        AddCustomAction(CENTURION_START_EVENT, 4000u, [&]() { HandleEventStart(); });
+        AddCustomAction(CENTURION_EVENT_RP, urand(2000, 10000), [&]() { HandleRP(); });
```

Comment:
This is incorrect, there is a Custom Action function prototype which accepts both min and max. This way every wipe it would use identical timer

Reply:
Should be correct now, or i missunderstood something :)